app/src/main/java/dev/shreyaspatil/foodium/ui/details/PostDetailsActivity.kt

```diff
@@ -87,27 +87,30 @@ class PostDetailsActivity : BaseActivity<PostDetailsViewModel, ActivityPostDetai
             }
 
             R.id.action_share -> {
-                val shareMsg = getString(
-                    R.string.share_message,
-                    post.title,
-                    post.author
-                )
-
-                val intent = ShareCompat.IntentBuilder.from(this)
-                    .setType("text/plain")
-                    .setText(shareMsg)
-                    .intent
-
-                if (intent.resolveActivity(packageManager) != null) {
-                    startActivity(intent)
-                }
+                showShareTray()
                 return true
             }
         }
 
         return super.onOptionsItemSelected(item)
     }
 
+    private fun showShareTray() {
+        val shareMsg = getString(
+                R.string.share_message,
+                post.title,
+                post.author
+        )
+        val intent = Intent(Intent.ACTION_SEND)
```

Comment:
Here we can use `ShareCompat` which was used earlier to generate intent. Please refer previous revision.